controller/tenant_minishift_test.go

```diff
@@ -0,0 +1,115 @@
+package controller_test
+
+import (
+	"context"
+	"fmt"
+	goatest "github.com/fabric8-services/fabric8-tenant/app/test"
+	"github.com/fabric8-services/fabric8-tenant/controller"
+	"github.com/fabric8-services/fabric8-tenant/tenant"
+	"github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/doubles"
+	"github.com/fabric8-services/fabric8-tenant/test/gormsupport"
+	"github.com/fabric8-services/fabric8-tenant/test/minishift"
+	"github.com/goadesign/goa"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"testing"
+	"time"
+)
+
+type TenantControllerMinishiftTestSuite struct {
+	minishift.TestSuite
+}
+
+func TestTenantControllerWithMinishift(t *testing.T) {
+	suite.Run(t, &TenantControllerMinishiftTestSuite{
+		TestSuite: minishift.TestSuite{DBTestSuite: gormsupport.NewDBTestSuite("../config.yaml")}})
+}
+
+func (s *TenantControllerMinishiftTestSuite) TestSetupUpdateCleanAndDeleteTenantNamespaces() {
+	// given
+	testdoubles.SetTemplateSameVersion("1abcd")
+	id := uuid.NewV4()
+	svc := goa.New("Tenants-service")
+	ctrl := controller.NewTenantController(svc, tenant.NewDBService(s.DB), s.GetClusterService(), s.GetAuthService(id), s.GetConfig())
+
+	// when setup is called
+	goatest.SetupTenantAccepted(s.T(), createUserContext(s.T(), id.String()), svc, ctrl)
+
+	// then
+	repo := tenant.NewDBService(s.DB)
+	tnnt, err := repo.GetTenant(id)
+	assert.NoError(s.T(), err)
+
+	iteration := 0
```

Comment:
are you trying to get the namespaces with a certain amount of time ? if so, you may want to use channels with a time out instead. You find some example in https://github.com/fabric8-services/fabric8-cluster/blob/master/application/transaction/transaction.go#L53-L84

Reply:
I think that the main difference is that I'm not running the iteration in goroutine.

Reply:
yes, indeed.

Reply:
my raw idea was to use 2 channels. One for the timeout, and another one that emits a message when the namespaces are created (with a loop to check at regular intervals - this needs to stay). But the function to test the availability of the channels could be externalized. does it make sense?

Reply:
this should be better I think - I hope that it helps: 0093df3